pkg/flow/caller.go

```diff
@@ -29,6 +29,13 @@ func NewRequest(functions functions.Stack, codec codec.Manager, metadata *metada
 	}
 }
 
+// Request represents a codec and metadata manager
+type Request struct {
```

Comment:
Just a question: what is the reason for making this struct public with private fields and no methods?
(if we make those field public the constructor is not needed anymore) :thinking:

Reply:
Looks indeed like a better approach. I will make the fields public and remove the constructor.